src/content/1.7/development/components/configuration.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Configuration storage
+weight: 80
+---
+
+# Configuration storage
+
+The Configuration storage of PrestaShop aims to address two needs.
+
+First, there is multiple situations where a single value has to be stored and be available easily. It can be a boolean to control order mode, it can be a limit for timeout: a simple key => value(s) storage is needed to manage multiple settings in PrestaShop.
+
+Secondly, modules might sometimes need a simple storage system, and they should not have to create a new SQL table for such a simple need.
+
+To address these needs, the Configuration storage of PrestaShop uses a SQL table `ps_configuration` and allows simple fetch, update and deletion of values in it.
+
+{{< figure src="/1.7/development/components/img/configuration_sql_structure.png" title="Configuration SQL structure" >}}
+
+Each value stored in the Configuration component can be controlled through given key, is compatible with Multishop (as it can be overriden for a shop or a group of shop) and has timestamps.
+
+The `Configuration` class is the interface allowing each module developer to use the component: to get or store data.
+
+## Usage
+
+### Store configuration data
+
+```php
+<?php
+Configuration::updateValue(string $key, mixed $value);
+```
+
+`$key` will be the reference to your data to reuse later.
+`$value` can be any scalar type (int, string, bool, array...).
```

Comment:
Most of the time array value means that configuration is multi-lang, we need to point this because if someone has a structure like:

```
0 => 'XYZ',
1 => 'ABC,
```

it will be treated as multi-lang even though the developer might not need it

Reply:
Also, an array is not a scalar:

> Scalar variables are those containing an int, float, string or bool. Types array, object and resource are not scalar.

https://www.php.net/manual/en/function.is-scalar.php

Reply:
Follow up from @kpodemski 's comment
```suggestion
`$value` can be any scalar type (int, float, string, bool) or an array.

{{% notice tip %}}
**Multi language**.

By convention, multilanguage values are stored in arrays indexed by language id, like this:

```php
[
    123 => 'Value in some language',
    456 => 'Value in some other language',
]
```
{{% /notice %}}
```